### ISSUE TITLE
Setting @current_user in authenticable.rb

### ISSUE THREAD

**Issue by hzalaz** [nsarno/knock#1]
In [authenticable.rb](https://github.com/nsarno/knock/blob/master/lib/knock/authenticable.rb#L8) the `@current_user` is retrieved like this

```ruby
@current_user = User.find(payload['user_id'])
```

but since there is no User model in the lib it will force whoever who uses knock to have a model named `User` and expect a claim named `user_id` to be present in the JWT.
What I'd like to have is a way to define how the user is retrieved from my DB, e.g. in my app's `config/knock.rb` I could have:

```ruby
config.current_user_from_token = -> (claims) { User.find(claims['sub']) }
```

and then in `authenticable.rb`

```ruby
@current_user = Knock.current_user_from_token.call(payload)
```

This will allow to fetch the user from any model using any claim in the JWT

**Comment by nsarno** [comment]
The gem in its original version is very opinionated as it's pulled from personal use cases. But I totally agree that it should evolve towards a more generic solution.

Thank you for your suggestions, I implemented them. This should resolve the issue.

**Comment by nsarno** [comment]
I released a new version of the gem (1.2.0)
